Amazon.QLDB.Driver.Tests/AsyncQldbDriverTests.cs

```diff
@@ -0,0 +1,361 @@
+                },
+                ResponseMetadata = new ResponseMetadata
+                {
+                    RequestId = "testId"
+                }
+            };
+            var sendCommandResponseStartTransaction = new SendCommandResponse
+            {
+                StartTransaction = new StartTransactionResult
+                {
+                    TransactionId = TestTransactionId
+                },
+                ResponseMetadata = new ResponseMetadata
+                {
+                    RequestId = "testId"
+                }
+            };
+            var sendCommandResponseExecute = new SendCommandResponse
+            {
+                ExecuteStatement = new ExecuteStatementResult
+                {
+                    FirstPage = new Page
+                    {
+                        NextPageToken = null,
+                        Values = ions
+                    }
+                },
+                ResponseMetadata = new ResponseMetadata
+                {
+                    RequestId = "testId"
+                }
+            };
+            var sendCommandResponseCommit = new SendCommandResponse
+            {
+                CommitTransaction = new CommitTransactionResult
+                {
+                    CommitDigest = new MemoryStream(h1.Hash),
+                    TransactionId = TestTransactionId
+                },
+                ResponseMetadata = new ResponseMetadata
+                {
+                    RequestId = "testId"
+                }
+            };
+        
+            mockClient.SetupSequence(x => x.SendCommandAsync(It.IsAny<SendCommandRequest>(), It.IsAny<CancellationToken>()))
+                    .Returns(Task.FromResult(sendCommandResponseWithStartSession))
+                    .Returns(Task.FromResult(sendCommandResponseStartTransaction))
+                    .Returns(Task.FromResult(sendCommandResponseExecute))
+                    .Returns(Task.FromResult(sendCommandResponseCommit));
+        
+            var driver = new AsyncQldbDriver(
+                new AsyncSessionPool((ct) => Session.StartSessionAsync("ledgerName", mockClient.Object, NullLogger.Instance, ct),
+                        AsyncQldbDriverBuilder.CreateDefaultRetryHandler(NullLogger.Instance), 4, NullLogger.Instance));
+
+            var result = await driver.ListTableNames();
+        
+            Assert.IsNotNull(result);
+            CollectionAssert.AreEqual(tables, result.ToList());
+        }
+        
+        [TestMethod]
+        public async Task TestAsyncExecuteWithActionLambdaCanInvokeSuccessfully()
+        {
+            var driver = new AsyncQldbDriver(
+                new AsyncSessionPool((ct) => Session.StartSessionAsync("ledgerName", mockClient.Object, NullLogger.Instance, ct),
+                    new Mock<IAsyncRetryHandler>().Object, 4, NullLogger.Instance));
+            await driver.Execute(txn => txn.Execute("testStatement"));
```

Comment:
Since the 'driver' is an AsyncQldbDriver and its Execute method overload accepts Func<AsyncTransactionExecutor, Task> action, then should not the lambda be declared as: async txn => await txn.Execute("testStatement")) ?